lib/src/components/sprite_animation_component.dart

```diff
@@ -43,8 +43,12 @@ class SpriteAnimationComponent extends PositionComponent {
     );
   }
 
+  /// Component will be removed after loop end and [removeOnFinish] is set.
+  /// [Component.shouldRemove] worked here.
   @override
-  bool get shouldRemove => removeOnFinish && (animation?.isLastFrame ?? false);
+  bool get shouldRemove =>
+      (removeOnFinish && (animation?.isLastFrame ?? false)) ||
```

Comment:
Nit: I think that `super.shouldRemove` should take precedence. I know that this don't change much since this is a `or` operation, but I think that it makes it easier to understand that logic since the super is more importante on this decision IMO

Reply:
Yes, done